### ISSUE TITLE
Challenge 47

### ISSUE THREAD

**Issue by bonham000** [bonham000/fcc-react-tests-module#133]
@HKuz I've merged the new challenge, everything looks great! I'm looking into writing the last test right now. Since you  have more recently actually worked through this sequence of challenges, I am wondering if it would be better to:

**a)** Leave 47 as it is, as a full repetition of 46, or
**b)** Write a new React component with far less code to really highlight the `map`ing part of it

What do you think?

For now I've left it as is and just modified the challenge with the test. The presence of the key can be checked with Enzyme in the component, however, it is mixed up in React's own react-id that it supplies (which may not be the same every time) so I just wrote a `.includes` to quickly check for the presence of `<li key = {` in the user's code, assuming that if that is there they have written the unique key.

**Comment by bonham000** [comment]
I'm also going to add a note in 46 that the next challenge covers adding the `key` just to not mislead anyone (be less misleading).

**Comment by HKuz** [comment]
@bonham000 - simplifying the challenge code is always good! Plus, it probably won't be as boring as solving the exact same solution, just with a `key` on the items. (Although, that did make writing the challenge easy...) I like your idea of really focusing on the `map` part, that's something they'll use a lot with React. We could tee up a component that already has an array of info available, then have them write out the entire `map` statement and render the new array.

Also, good call on the note in 46.

**Comment by bonham000** [comment]
Ok! I agree with you. I will try to write a simplified challenge for this, and I think I am also going to add that rendering styles conditionally challenge.

**Comment by HKuz** [comment]
NICE - yeah, conditional styles will be a good lesson. I'm also curious to see how the key test works - I got the mocked component simulation to work, and could access the `li`s, but couldn't see any `key` prop or attribute on them. I figure there's a different/better approach for that.

**Comment by bonham000** [comment]
@HKuz — Okay! I've added 45 and re-drafted 47. I think these are good to go but you will probably want to give them one more quick review just to be sure. Also, feel free to check out the tests in 47 for the unique `key`, I did go back and actually check to be sure the keys would be different (at least I checked for two elements anyway). The key attribute only appears in the unique ID react gives the element, as far as I can tell. I think the only way you'll pass that challenge is through the way we are looking for. And now...

**50**

⭐️ ⭐️ ⭐️
